libraries/AP_HAL_ChibiOS/Storage.cpp

```diff
@@ -44,64 +44,72 @@ extern const AP_HAL::HAL& hal;
 
 void Storage::_storage_open(void)
 {
-    if (_initialised) {
+    if (_initialisedType != StorageBackend::None) {
         return;
     }
 
-#ifdef USE_POSIX
-    // if we have failed filesystem init don't try again
-    if (log_fd == -1) {
-        return;
-    }
-#endif
-
     _dirty_mask.clearall();
 
 #if HAL_WITH_RAMTRON
-    using_fram = fram.init();
-    if (using_fram) {
-        if (!fram.read(0, _buffer, CH_STORAGE_SIZE)) {
+    if (fram.init()) {
+        if (fram.read(0, _buffer, CH_STORAGE_SIZE) == CH_STORAGE_SIZE) {
+            _save_backup();
+            _initialisedType = StorageBackend::FRAM;
+            hal.console->printf("Initialised Storage type=%d\n", _initialisedType);
             return;
         }
-        _save_backup();
-        _initialised = true;
-        return;
     }
+
+    #if defined(HAL_RAMTRON_NO_FALLBACK) && HAL_RAMTRON_NO_FALLBACK
```

Comment:
I'd rather have HAL_RAMTROM_ALLOW_FALLBACK and not fallback by default. See #13581

Reply:
Ack. I was attempting to make this change have minimal impact on other board types.

I've inverted the name (and logic) to `HAL_RAMTROM_ALLOW_FALLBACK`. Are you comfortable with the potential breakage on boards with misconfigured RAMTRON devices? Would it be better to add a default value of `true` for this define, and swap it over to 'false' once others have had some time to check their boards?